message Track {
	optional string id = 2; //i dont know what this is
	optional int32 creation = 3; //unix timestamp seconds - only a guess
	optional int32 lastPlayed = 4; //unix timestamp seconds - only a guess
	optional string title = 6; //fairly certain
	optional string artist = 7; //fairly certain
	optional string composer = 8; //seems to be exclusively ""
	optional string album = 9; //seems to be exclusively ""
	optional string albumArtist = 10; //seems to be exclusively ""
	optional int32 year = 11; //seems to be exclusively 0
	optional string u4 = 12; //seems to be exclusively ""
	optional int32 track = 13; //seems to be exclusively 0
	optional string genre = 14;	//fairly certain
	optional int32 duration = 15; //fairly certain
	optional int32 u6 = 16; //seems to be exclusively 0
	optional int32 u7 = 20; //seems to be exclusively 0
	optional int32 totalTracks = 26; //seems to be exclusively 0
	optional int32 u9 = 27; //seems to be exclusively 0
	optional int32 u10 = 28; //seems to be exclusively 0
	optional int32 u11 = 31; //seems to be exclusively 1
	optional int32 u12 = 32; //values include 5115029, 10396779, 5923343, 7640338, 8059133
	optional int32 u13 = 37; //seems to be exclusively 0
	optional int32 u14 = 38; //seems to be exclusively 0
	optional int32 bitrate = 44; //mostly certain
	optional string u15 = 53; //seems to be exclusively "-383260437"
	optional int32 u16 = 61; 
}


message MetadataRequest {
	repeated Track tracks = 1;
	optional string address = 2; //I don't know what this is
}

message QueuedUpload {
	optional string id = 1;
	optional int32 u0 = 2; //i dont know but this is always 2
	optional string serverId = 3;  //That's what this number is referred to later as during the uploadsj/rupio
}

message Status {
	optional int32 u0 = 1;
	optional int32 u1 = 2;
	optional int32 u2 = 3; //like totally always five
	optional int32 u3 = 4; //this is like totally always 6000
	optional int32 u4 = 5; 
	optional int32 u5 = 6; 
}


message TrackResponse {
	repeated string ids = 2;
	repeated QueuedUpload uploads = 3;	
}

message MetadataResponse {
	optional int32 u0 = 1; //no idea what this is, but the value is 1
	optional TrackResponse response = 2;
	optional Status state = 6;
}

message UploadAuth {
	optional string address = 1;
	optional string hostname = 2;
}

message UploadAuthResponse {
	optional int32 u0 = 1;
	optional Status status = 6;
	optional int32 u1 = 11;
	optional int32 u2 = 12;
}